### PR TITLE
[fix](join) incorrect result of null aware left anti join with empty build side

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

In a null aware left anti join, if the right side is empty, all rows from the left side should be output, regardless of whether the row is null or not.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/common/hash_table/hash_map.h [hunk 1]
```diff
@@ -338,6 +341,26 @@ class JoinHashMapTable : public HashMapTable<Key, Cell, Hash, Grower, Allocator>
         auto matched_cnt = 0;
         const auto batch_size = max_batch_size;
 
+        if (_empty_table) {
```
Comment (BiteTheDDDDt) [inline be/src/vec/common/hash_table/hash_map.h:344]: add a function to process case when left anti+empty

File: be/src/vec/common/hash_table/hash_map.h [hunk 2]
```diff
@@ -338,6 +341,26 @@ class JoinHashMapTable : public HashMapTable<Key, Cell, Hash, Grower, Allocator>
...
+                    build_idxs[matched_cnt] = 0;
+                    matched_cnt++;
+                }
+
+                if constexpr (!with_other_conjuncts) {
+                    mark_column->resize_fill(matched_cnt,
+                                             JoinOpType != doris::TJoinOp::LEFT_SEMI_JOIN);
+                }
+            } else {
+                LOG(FATAL) << "CROSS JOIN SHOULD NOT BUILD HASH TABLE";
```
Comment (BiteTheDDDDt) [inline be/src/vec/common/hash_table/hash_map.h:359]: do not log fatal

File: be/src/vec/common/hash_table/hash_map.h [hunk 3]
```diff
@@ -338,6 +341,26 @@ class JoinHashMapTable : public HashMapTable<Key, Cell, Hash, Grower, Allocator>
         auto matched_cnt = 0;
         const auto batch_size = max_batch_size;
 
+        if (_empty_table) {
+            if constexpr (JoinOpType == doris::TJoinOp::LEFT_ANTI_JOIN ||
+                          JoinOpType == doris::TJoinOp::NULL_AWARE_LEFT_ANTI_JOIN ||
+                          JoinOpType == doris::TJoinOp::LEFT_SEMI_JOIN) {
```
Comment (BiteTheDDDDt) [inline be/src/vec/common/hash_table/hash_map.h:347]: does left semi need to match?

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**